src/app/ReadClient.cpp

```diff
@@ -756,11 +706,14 @@ CHIP_ERROR ReadClient::ProcessEventReportIBs(TLV::TLVReader & aEventReportIBsRea
 CHIP_ERROR ReadClient::RefreshLivenessCheckTimer()
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
+
     CancelLivenessCheckTimer();
-    VerifyOrReturnError(mpExchangeCtx != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
-    VerifyOrReturnError(mpExchangeCtx->HasSessionHandle(), err = CHIP_ERROR_INCORRECT_STATE);
 
-    System::Clock::Timeout timeout = System::Clock::Seconds16(mMaxInterval) + mpExchangeCtx->GetSessionHandle()->GetAckTimeout();
+    VerifyOrReturnError(mExchange, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mExchange->HasSessionHandle(), err = CHIP_ERROR_INCORRECT_STATE);
```

Comment:
```suggestion
    VerifyOrReturnError(mExchange, CHIP_ERROR_INCORRECT_STATE);
    VerifyOrReturnError(mExchange->HasSessionHandle(), CHIP_ERROR_INCORRECT_STATE);
```
I know, it was pre-existing....